diff --git a/dist/index.d.ts b/dist/index.d.ts
index f17e71112bbddff8f2271d804d896bb502e9959a..b40c8988b95897f9630e21b02aba7faaa31d2f58 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -226,7 +226,7 @@ export interface FormApi<FormValues = Record<string, any>, InitialFormValues = P
   ) => void
   submit: () => Promise<FormValues | undefined> | undefined
   subscribe: (
-    subscriber: FormSubscriber<FormValues>,
+    subscriber: FormSubscriber<FormValues, InitialFormValues>,
     subscription: FormSubscription
   ) => Unsubscribe
 }
