diff --git a/dist/index.d.ts b/dist/index.d.ts
index dc9439a408e1108e51cfbf867a83f945bbf8e484..857d3a8954d72c33ebfc659ee6734cebc350a29e 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -237,7 +237,7 @@ export interface FormApi<
   ) => void;
   submit: () => Promise<FormValues | undefined> | undefined;
   subscribe: (
-    subscriber: FormSubscriber<FormValues>,
+    subscriber: FormSubscriber<FormValues, InitialFormValues>,
     subscription: FormSubscription,
   ) => Unsubscribe;
 }
