diff --git a/dist/index.d.ts b/dist/index.d.ts
index db1e0e2b8f26ee5829f3256f4c11c41eb8ed05bb..84dca0b987638cfa54fbf55363ca8095cc07d858 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -237,7 +237,7 @@ export interface FormApi<
   ) => void;
   submit: () => Promise<FormValues | undefined> | undefined;
   subscribe: (
-    subscriber: FormSubscriber<FormValues>,
+    subscriber: FormSubscriber<FormValues, InitialFormValues>,
     subscription: FormSubscription,
   ) => Unsubscribe;
 }
