diff --git a/dist/index.d.ts b/dist/index.d.ts
index 617484f0aca73966fb7a57cdf039f1c77a5200a5..700051b103b5a633492838d67c0453f2a078cf44 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -237,7 +237,7 @@ export interface FormApi<
   ) => void;
   submit: () => Promise<FormValues | undefined> | undefined;
   subscribe: (
-    subscriber: FormSubscriber<FormValues>,
+    subscriber: FormSubscriber<FormValues, InitialFormValues>,
     subscription: FormSubscription,
   ) => Unsubscribe;
 }
