diff --git a/index.js b/index.js
index e6bd17ff5b3669e1ad6ba4384f64ac777c692ae5..c8d16ef48445896335b37d0046be9a6f6f5e151b 100644
--- a/index.js
+++ b/index.js
@@ -1124,7 +1124,7 @@ function astFromEnumValue(value, schema, pathToDirectivesInExtensions) {
             value: value.name,
         },
         // ConstXNode has been introduced in v16 but it is not compatible with XNode so we do `as any` for backwards compatibility
-        directives: getDirectiveNodes(value, schema, pathToDirectivesInExtensions),
+        directives: getDeprecatableDirectiveNodes(value, schema, pathToDirectivesInExtensions),
     };
 }
 function makeDeprecatedDirective(deprecationReason) {
