diff --git a/index.js b/index.js
index 617b087496485e34664775eccce763606cb7aa2d..436cdd1b73456ca9bec554051e36f4b0eb03e0ce 100644
--- a/index.js
+++ b/index.js
@@ -127,6 +127,7 @@ function hasNullableTypeRecursively(type) {
     return false;
 }
 function isUsingTypes(document, externalFragments, schema) {
+    return true;
     let foundFields = 0;
     const typesStack = [];
     graphql.visit(document, {
