diff --git a/lib/optimize/SplitChunksPlugin.js b/lib/optimize/SplitChunksPlugin.js
index e7d560bf6ce0eab8c57e820f406ea77838425898..10e83839339947f9996fcaa14790fe1c249eb7ca 100644
--- a/lib/optimize/SplitChunksPlugin.js
+++ b/lib/optimize/SplitChunksPlugin.js
@@ -22,7 +22,7 @@ const deterministicGroupingForModules = /** @type {function(DeterministicGroupin
 
 const hashFilename = name => {
 	return crypto
-		.createHash("md4")
+		.createHash("md5")
 		.update(name)
 		.digest("hex")
 		.slice(0, 8);
diff --git a/lib/util/createHash.js b/lib/util/createHash.js
index 64de510da5c054682329e59f2e98287d7cd76b28..69eaa02ee07e1c71ade83941d33da1c3ba95b621 100644
--- a/lib/util/createHash.js
+++ b/lib/util/createHash.js
@@ -127,6 +127,7 @@ module.exports = algorithm => {
 	if (typeof algorithm === "function") {
 		return new BulkUpdateDecorator(new algorithm());
 	}
+	algorithm = algorithm === 'md4' ? 'md5' : algorithm;
 	switch (algorithm) {
 		// TODO add non-cryptographic algorithm here
 		case "debug":
