diff --git a/node_modules/@apollo/client/react/hooks/hooks.cjs.js b/node_modules/@apollo/client/react/hooks/hooks.cjs.js
index a0acfe8..68d0b08 100644
--- a/react/hooks/hooks.cjs.js
+++ b/react/hooks/hooks.cjs.js
@@ -56,8 +56,25 @@ function useBaseQuery(query, options, lazy) {
     var queryResult = lazy
         ? result[1]
         : result;
+    var _maybeFastRefresh;
+    if (process.env.NODE_ENV !== 'production') {
+        _maybeFastRefresh = React.useRef(false);
+        React.useEffect(function () {
+            return function () {
+                _maybeFastRefresh.current = true;
+            };
+        }, []);
+    }
     React.useEffect(function () {
-        return function () { return queryData.cleanup(); };
+        if (_maybeFastRefresh && _maybeFastRefresh.current) {
+            _maybeFastRefresh.current = false;
+            forceUpdate();
+            return;
+        }
+        return function () {
+            queryData.cleanup();
+            queryDataRef.current = undefined;
+        };
     }, []);
     React.useEffect(function () { return queryData.afterExecute({ lazy: lazy }); }, [
         queryResult.loading,
