diff --git a/dist/index.d.ts b/dist/index.d.ts
index dbcf3379..cd7907f3 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -5,7 +5,7 @@ export interface AnyObject {
   [key: string]: any
 }
 export interface ValidationErrors extends AnyObject {}
-export interface SubmissionErrors extends AnyObject {}
+export type SubmissionErrors = AnyObject | undefined;
 
 export interface FormSubscription {
   active?: boolean
@@ -224,7 +224,7 @@ export interface FormApi<FormValues = Record<string, any>, InitialFormValues = P
   ) => void
   submit: () => Promise<FormValues | undefined> | undefined
   subscribe: (
-    subscriber: FormSubscriber<FormValues>,
+    subscriber: FormSubscriber<FormValues, InitialFormValues>,
     subscription: FormSubscription
   ) => Unsubscribe
 }
