diff --git a/index.js b/index.js
index f326eeaf71fa2b4df15b1bf485abc0db0fc7331b..f4bab182a76f0dddffc8ef2e12cbc06c9de82e5d 100644
--- a/index.js
+++ b/index.js
@@ -1124,7 +1124,7 @@ function astFromEnumValue(value, schema, pathToDirectivesInExtensions) {
             value: value.name,
         },
         // ConstXNode has been introduced in v16 but it is not compatible with XNode so we do `as any` for backwards compatibility
-        directives: getDirectiveNodes(value, schema, pathToDirectivesInExtensions),
+        directives: getDeprecatableDirectiveNodes(value, schema, pathToDirectivesInExtensions),
     };
 }
 function makeDeprecatedDirective(deprecationReason) {
