diff --git a/types/index.d.ts b/types/index.d.ts
index b756e3a3..dd2c0cda 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -151,6 +151,7 @@ export interface Components {
   Container?: React.ComponentType<any>;
   EditField?: React.ComponentType<any>;
   EditRow?: React.ComponentType<any>;
+  EditCell?: React.ComponentType<any>;
   FilterRow?: React.ComponentType<any>;
   Groupbar?: React.ComponentType<any>;
   GroupRow?: React.ComponentType<any>;
@@ -177,23 +178,23 @@ export const MTableToolbar: (props: any) => React.ReactElement<any>;
 export const MTable: (props: any) => React.ReactElement<any>;
 
 export interface Icons {
-  Add?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Check?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Clear?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Delete?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  DetailPanel?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Edit?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Export?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Filter?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  FirstPage?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  SortArrow?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  LastPage?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  NextPage?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  PreviousPage?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  ResetSearch?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  Search?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  ThirdStateCheck?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
-  ViewColumn?: React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>;
+  Add?: SvgIconComponent;
+  Check?: SvgIconComponent;
+  Clear?: SvgIconComponent;
+  Delete?: SvgIconComponent;
+  DetailPanel?: SvgIconComponent;
+  Edit?: SvgIconComponent;
+  Export?: SvgIconComponent;
+  Filter?: SvgIconComponent;
+  FirstPage?: SvgIconComponent;
+  SortArrow?: SvgIconComponent;
+  LastPage?: SvgIconComponent;
+  NextPage?: SvgIconComponent;
+  PreviousPage?: SvgIconComponent;
+  ResetSearch?: SvgIconComponent;
+  Search?: SvgIconComponent;
+  ThirdStateCheck?: SvgIconComponent;
+  ViewColumn?: SvgIconComponent;
 }
 
 export interface Options {
