diff --git a/typings/react-dropzone.d.ts b/typings/react-dropzone.d.ts
index ca436931..29131f03 100644
--- a/typings/react-dropzone.d.ts
+++ b/typings/react-dropzone.d.ts
@@ -48,12 +48,12 @@ export type DropzoneState = DropzoneRef & {
   fileRejections: FileRejection[];
   rootRef: React.RefObject<HTMLElement>;
   inputRef: React.RefObject<HTMLInputElement>;
-  getRootProps(props?: DropzoneRootProps): DropzoneRootProps;
-  getInputProps(props?: DropzoneInputProps): DropzoneInputProps;
+  getRootProps: (props?: DropzoneRootProps) => DropzoneRootProps;
+  getInputProps: (props?: DropzoneInputProps) => DropzoneInputProps;
 };

 export interface DropzoneRef {
-  open(): void;
+  open: () => void;
 }

 export interface DropzoneRootProps extends React.HTMLAttributes<HTMLElement> {
